### ISSUE TITLE
DNS records created through programmable-multiple relations aren't deleted on departure

### ISSUE THREAD

**Issue by mirw** [chuckbutler/DNS-Charm#8]
I'm using the DNS charm's programmable-multiple interface as part of the Clearwater Juju Charms (at https://github.com/Metaswitch/clearwater-juju/tree/dnsaas).

When nodes add a relation to the DNS charm, DNS records are created.  However, on departure, they aren't deleted.

This causes an issue for elastic scale-up/scale-down as, after scale-down (and units departing the relation), their records are still present in DNS, so we try to load-balance across a large set of dead nodes.

It looks as though the programmable-multiple-relation-departed hook isn't implemented.  Are there any plans to add this?

**Comment by chuckbutler** [comment]
@mirw you are correct, the departed relationship was not implemented.  There's an active thread going on over email with me about these issues, and I wont be able to look into this directly for at least 2 weeks. If i can squeeze in some hot fixes before then I'll make sure to ping w/ the branch for testing.

Ideally as the primary consumer of this charm I'd like to get your validation of any features/fixes.

I will reach out as soon as I have a hotfix for the charm.

**Comment by mirw** [comment]
Thanks for the update!  That sounds good - yes, please let me know if/when you have something I can validate.

**Comment by chuckbutler** [comment]
@mirw I've got PR #16 prepped to address this issue. Can you run a validation for me and ensure this is behaving as you would expect before I merge it in?

**Comment by mirw** [comment]
@rkd-msw, please can you take a look at this one too?
